src/count-islands/index.js

```diff
@@ -1,3 +1,38 @@
-function countIslands(grid) {}
+function countIslands(grid) {
+  const rowLength = grid.length;
+  const colLength = grid[0].length;
+  let islands = 0;
+  const search = (currentRow, currentCol) => {
```

Comment:
move the search() function out of the countIsLands() function. A function should have a single responsibility that way your code is easily maintainable.